cunumeric/sphinxext/_comparison_config.py

```diff
@@ -0,0 +1,404 @@
+# Copyright 2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from __future__ import annotations
+
+from dataclasses import dataclass
+from types import (
+    BuiltinFunctionType,
+    FunctionType,
+    MethodDescriptorType,
+    MethodType,
+)
+
+import numpy
+
+__all__ = (
+    "GROUPED_CONFIGS",
+    "MISSING_NP_REFS",
+    "NUMPY_CONFIGS",
+    "SKIP",
+)
+
+SKIP = {
+    "abs",
+    "add_docstring",
+    "add_newdoc_ufunc",
+    "add_newdoc",
+    "alen",
+    "alltrue",
+    "bitwise_not",
+    "compare_chararrays",
+    "cumproduct",
+    "fastCopyAndTranspose",
+    "get_array_wrap",
+    "iterable",
+    "max",
+    "min",
+    "ndim",
+    "product",
+    "recfromcsv",
+    "recfromtxt",
+    "round",
+    "safe_eval",
+    "set_numeric_ops",
+    "size",
+    "sometrue",
+    "test",
+}
+
+# these do not have valid intersphinx references
+MISSING_NP_REFS = {
+    "numpy.loads",
+    "numpy.mafromtxt",
+    "numpy.ndarray.flat",
+    "numpy.ndarray.shape",
+    "numpy.ndarray.T",
+    "numpy.ndfromtxt",
+}
+
+
+@dataclass(frozen=True)
+class SectionConfig:
+    title: str
+    attr: str | None
```

Comment:
Just to confirm, has this union syntax become available since we bump up the minimum python version to 3.8?

Reply:
No that was added in https://docs.python.org/3/whatsnew/3.10.html#pep-604-new-type-union-operator we also still need `__future__` for using builtins for typing e.g. `list[int]` instead of `List[int]`

Reply:
I guess we were going to use `Union` event though the `__future__` import brings them in. FWIW the pipe syntax has crept in elsehwhere:
```
legate/core/io.py
106:        tile_shape : int | Tuple[int]
150:    shape : int | Tuple[int]
153:    colors : int | Tuple[int]
168:    get_local_colors : Callable[[], Iterable[Point]] | None

legate/core/_legion/operation.py
1209:        field : int | FieldID

legate/core/_legion/partition_functor.py
349:        domains : FutureMap | dict[Point, Rect]
```
Do you want a change here and a PR to avoid this?

Reply:
No we don't have to avoid it, but if we're gonna use it, I just want us to use it everywhere. I believe the pipe syntax can be used only in type annotations, right? Unless we missed out any places where it could be used but wasn't, I think this is fine.